0.8.0 branch

praveksharma opened the pull request:
<!-- Please give a brief explanation of the purpose of this pull request. -->

Please see the relevant pre-release (currently in draft):  https://github.com/open-quantum-safe/oqs-provider/releases/tag/untagged-e8b2d0985929c684a8d5

This PR makes the following changes:

- Bumps the version string, 
- stores a copy of `generate.yml` as `generate.yml-0.12.0`, 
- Updates `SECURITY.md`, 
- Updates `RELEASE.md`, 
- Points CI to liboqs 0.12.0, and 
- Points CI to OpenSSL 3.4.0.

Checks done:

- Stable under `generate.py` and `do_code_format.sh`,
- CI is green,
- 'LIBOQS_BRANCH=0.12.0 OPENSSL_BRANCH=openssl-3.4.0 ./scripts/fullbuild.sh -F && ./scripts/runtests.sh -V` runs OK,
- `release-tests.sh` with liboqs 0.12.0 and openssl 3.4.0 runs OK,
- nginx, curl, and httpd docker images build OK, and
- OpenSSL external test integration passes when run locally with [`gh act --job external-tests-providers`](https://github.com/openssl/openssl/blob/79c9cbbe1f9c3b8314312b6d8bb25b7138831e04/.github/workflows/ci.yml#L623C12-L623C55). (Logs attached: [logs.txt](https://github.com/user-attachments/files/18135527/logs.txt). Note that the external-tests-providers jobs fails because the tests for pkcs11-provider fail but oqs-provider passes.).


This PR will not land; once approved, a new 0.8.0-release PR will be made. 

<!-- Does this PR resolve any issue?  If so, please reference it using automatic-closing keywords like "Fixes #123." -->

<!-- Once your pull request is ready for review and passing continuous integration tests, please convert from a draft PR to a normal PR, and request a review from one of the OQS core team members. -->


<!-- Does this PR resolve any issue?  If so, please reference it using automatic-closing keywords like "Fixes #123." -->

<!-- Once your pull request is ready for review and passing continuous integration tests, please convert from a draft PR to a normal PR, and request a review from one of the OQS core team members. -->

baentsch commented on RELEASE.md:
```diff
@@ -14,7 +14,47 @@ Further details on building, testing and use can be found in [README.md](https:/
 ## Release notes
 
-This is version 0.7.1-dev of oqs-provider which continues from the earlier 0.7.0 release. This release is fully tested to be used in conjunction with the main branch of [liboqs](https://github.com/open-quantum-safe/liboqs) and is guaranteed to be in sync with v0.12.0 of `liboqs` as and when released.
+This is version 0.8.0-rc1 of oqs-provider which continues from the earlier 0.7.0 release. This release is fully tested to be used in conjunction with the main branch of [liboqs](https://github.com/open-quantum-safe/liboqs) and is guaranteed to be in sync with v0.12.0 of `liboqs` as and when released.
```
nit: suggest deletion of trailing "as and when released".

baentsch commented on RELEASE.md:
```diff
@@ -14,7 +14,47 @@ Further details on building, testing and use can be found in [README.md](https:/
+
+### Deprecation notice
+
+The associated liboqs v0.12.0 release will be the last liboqs release to include Kyber (that is, the NIST Round 3 version of Kyber, prior to its standardization by NIST as ML-KEM in FIPS 203). Applications should switch to ML-KEM (FIPS 203).
```
Why is this phrased in `liboqs` terms? This is `oqsprovider` and the ramifications for _its_ users should be documented, e.g., "This is to notify users of *kyber* algorithms to switch to the standardized "*mlkem*" algorithms as support for Kyber will be removed with the next release of `liboqs` and `oqsprovider`."

Presence/absence of APIs invisible/inaccessible to users of `oqsprovider` IMO do not need to be documented at all in the release notes.

baentsch commented on RELEASE.md:
nit: typo.

baentsch replied:
In addition: The context string support in `oqsprovider` is a (new) feature entirely independent of ML-DSA and warrants its own line item.

praveksharma commented:
Thank you for the suggestions @baentsch! I've included the changes here and on the draft pre-release: https://github.com/open-quantum-safe/oqs-provider/releases/tag/untagged-0fb2ad16a1ba976ea75a

baentsch commented on RELEASE.md:
Well, also those for Frodo, obviously. Although those are not IANA code points, strictly speaking :)

praveksharma replied:
Thank you pointing this out @baentsch! I copied the PR title over from #577 without inspecting file level changes. Could you please suggest a more suitable description?

baentsch replied:
What about "updates IANA code points for ML-KEM also changing Frodo code points"? The latter are not IANA approved but at least now are in a reserved space.

praveksharma commented on RELEASE.md:
```diff
@@ -14,7 +14,48 @@ Further details on building, testing and use can be found in [README.md](https:/
+
+This is to notify users of Kyber (Round 3 version) to switch to the ML-KEM (FIPS 203 final version) as support for Kyber will be removed with the next release of liboqs and oqsprovider.
+
+The addition of ML-DSA FIPS 204 final version to liboqs v0.12.0 has introduced a new signature API which includes a context string parameter. The liboqs team is planning to remove the old version of the API without a context string in the next release to streamline the API and bring it in line with NIST specifications. Users who have an opinion on this removal are invited to provide input at [lib ...
```
Noted, I shall remove line 23 entirely.

baentsch commented on RELEASE.md:
Suggest re-phrasing to be easier to read "Adds support for context strings in OpenSSL versions >= 3.2"

baentsch commented on RELEASE.md:
DTLS1.3 support also is noteworthy in my opinion, e.g., "Adds support for DTLS1.3". In this case, it's for a feature branch only but no code changes will be required as and when this gets enabled in OpenSSL, so not sure how to phrase that restriction (as outside our control).

SWilson4 replied:
Perhaps "Adds support for DTLS 1.3 (pending support in OpenSSL)"?

baentsch replied:
"pending support" sounds "too pessimistic"/as if this is not available -- which it kind of is: If you build the feature branch you have it (though admittedly not yet fully finished)... I also thought about "(if enabled)" but that's also not quite cutting it ("too optimistic").... Given (only?) those options, though, I'd go with yours @SWilson4 : It's more cautious and hence, more appropriate.

ghen2 replied:
"requires OpenSSL 3.5" is clearer and may age better? (assuming it will be part of that release)

praveksharma replied:
If there is consensus I shall proceed with "Adds support for DTLS 1.3 when built against OpenSSL versions >= 3.5. Of course, only if OpenSSL 3.5 is the release which will support this feature; I couldn't learn much with cursory search, could you please confirm this @baentsch? If the exact release is uncertain at the moment I shall proceed with Spencer's suggestion.

baentsch replied:
I don't think this is guaranteed to be included in 3.5. What's your take @fwh-dc ? As I said above, I also support @SWilson4 's wording.

fwh-dc replied:
I support @SWilson4's wording. I wouldn't mention Openssl version numbers before it is set in stone.

praveksharma commented:
@baentsch @SWilson4 @ghen2 @fwh-dc Thank you for you suggestions, I've updated README.md to include them and updated the draft pre-release as well: https://github.com/open-quantum-safe/oqs-provider/releases/tag/untagged-d43fdf73a30644bc1820

baentsch reviewed (approved):
Thx!

baentsch commented:
Following guidance in https://github.com/open-quantum-safe/oqs-provider/wiki/Release-process this removed "rc1" tags and waits for approval to land. Final release will be cut on the landed (squashed) commit.

praveksharma commented:
> Following guidance in https://github.com/open-quantum-safe/oqs-provider/wiki/Release-process this removed "rc1" tags and waits for approval to land. Final release will be cut on the landed (squashed) commit.

Thank you for finishing up the final release steps @baentsch! The pre-release is still in draft, is that intentional? Please let me know if that is becuase the GitHub web UI won't let you publish it and I shall do the same.

I'm linking a discussion related to the 0.7.0 release to see if we'd like to wait for a little bit before proceeding with the final release after appropriate approvals since I have not been able to automate the release process yet: https://github.com/open-quantum-safe/oqs-provider/pull/534#discussion_r1796918305

baentsch commented:
Thanks for the reminder to move rc1 out of Draft, @praveksharma . Just done. And sure, 0.8.0 will sit un-merged for a day before I squash, commit, and release (as agreed in the call yesterday) - just in case :)

baentsch commented:
Looking at https://github.com/open-quantum-safe/boringssl/pull/130 I wonder whether we should also at least announce removal of Dilithium with the next release (along the lines of Kyber)?

fwiw, I very much like the approach taken by @pi-314159 for oqs-boringssl: Just yank the stuff that's outdated... Or maybe it's just envy :-) @open-quantum-safe/oqs-committers do you know of anyone still relying on Kyber and Dilithium still being supported? May it be an idea to mark them "disabled" _already now_ in the generate.yml? This would functionally align oqsprovider and oqsboringssl and still allow anyone to enable those algs again if they absolutely must.

dstebila commented:
I've updated openquantumsafe.org with info about the release.
